src/app/screens/Accounts/ImportSecretKey/index.tsx

```diff
@@ -13,31 +13,35 @@ import MnemonicInputs from "~/app/components/mnemonic/MnemonicInputs";
 import api from "~/common/lib/api";
 
 function ImportSecretKey() {
-  const [mnemonic, setMnemonic] = useState<string>("");
   const { t: tCommon } = useTranslation("common");
   const navigate = useNavigate();
   const { t } = useTranslation("translation", {
     keyPrefix: "accounts.account_view.mnemonic",
   });
 
-  const [hasFetchedData, setHasFetchedData] = useState(false);
+  const [mnemonic, setMnemonic] = useState<string>("");
+  const [loading, setLoading] = useState<boolean>(true);
   const [hasNostrPrivateKey, setHasNostrPrivateKey] = useState(false);
+
   const { id } = useParams() as { id: string };
 
   useEffect(() => {
     (async () => {
       try {
+        setLoading(true);
+
         const account = await api.getAccount(id);
         setHasNostrPrivateKey(account.nostrEnabled);
         if (account.hasMnemonic) {
           // do not allow user to import a mnemonic if they already have one for the current account
           // go to account settings
           navigate(`/accounts/${id}`);
         }
-        setHasFetchedData(true);
       } catch (e) {
         console.error(e);
         if (e instanceof Error) toast.error(`Error: ${e.message}`);
+      } finally {
+        setLoading(false);
```

Comment:
what happens if the catch block occurs? it might be better to not set loading to false in that case